xbmc/windowing/wayland/WinSystemWaylandWebOS.cpp

```diff
@@ -74,6 +93,9 @@ CWinSystemWaylandWebOS::~CWinSystemWaylandWebOS() noexcept
     m_webosRegistry->UnbindSingletons();
   }
   m_webosRegistry.reset();
+
+  if (m_requestContext)
+    HUnregisterServiceCallback(m_requestContext.get());
```

Comment:
make HUnregisterServiceCallback the deleter of the m_requestContext unique_ptr; this should be preferred to custom cleanup and here it's a rather simple case :)